### PR TITLE
Adds an init file for SUSE

### PR DESCRIPTION
**Pull request by gmmeyer**
The current init files do not work for suse, they all have various errors. I took the Ubuntu one, added in the missing functions that were throwing up the errors (I had to include the copyright text as well).

This is not the best and I fully admit that there are problems with this approach, but it's better than any of the other solutions I considered: rewriting the init file would be a time consuming and error prone task--I am certain that these functions work and that our current init file works.

The Red Hat file has similar issues. I just picked the debian one because it was easy to fix.

The systemd file works fine, so no such replacement is needed.

### CODE REVIEW

File: packaging/suse/datadog-agent.init [hunk 1]
```diff
@@ -0,0 +1,481 @@
...
+        done
+        # After 10 tries the agent didn't start. Report an error
+        log_end_msg 1
+        check_status # report what went wrong
+        $0 stop
+        exit 1
+        ;;
+    stop)
+        log_daemon_msg "Stopping $DESC (stopping supervisord)" "$NAME"
+        /sbin/start-stop-daemon --stop --quiet --oknodo --pidfile $SUPERVISOR_PIDFILE
```
Comment (truthbk) [inline packaging/suse/datadog-agent.init]: I noticed the `debian` counterpart had a `--retry 30` argument for `start-stop-daemon` is that not available in suse?

File: packaging/suse/datadog-agent.init [hunk 2]
```diff
@@ -0,0 +1,481 @@
...
+
+        su $AGENTUSER -c "$AGENTPATH configcheck" > /dev/null 2>&1
+        if [ $? -ne 0 ]; then
+            log_daemon_msg "Invalid check configuration. Please run sudo /etc/init.d/datadog-agent configtest for more details."
+            log_daemon_msg "Resuming starting process."
+        fi
+
+
+        log_daemon_msg "Starting $DESC (using supervisord)" "$NAME"
+        PATH=$SYSTEM_PATH /sbin/start-stop-daemon --start --quiet --oknodo --exec $SUPERVISORD_PATH -- -c $SUPERVISOR_FILE --pidfile $SUPERVISOR_PIDFILE
```
Comment (truthbk) [inline packaging/suse/datadog-agent.init]: Pretty crazy the PATH isn't enforced despite putting `/sbin` in it at the top of the script.

Comment (gmmeyer) [inline packaging/suse/datadog-agent.init]: This was in the debian script, I believe, but yes it is.

File: packaging/suse/datadog-agent.init (hunk 1 above)
Comment (gmmeyer) [inline packaging/suse/datadog-agent.init]: This is not available, unfortunately

Comment (gmmeyer) [inline packaging/suse/datadog-agent.init]:
```
greg@linux-tnwu:~/Desktop> /sbin/start-stop-daemon --help
start-stop-daemon for Debian Linux - small and fast C version written by
Marek Michalkiewicz <marekm@i17linuxb.ists.pwr.wroc.pl>, public domain.
version 0.3.1, 1996-07-19

Usage:
    start-stop-daemon -S|--start options ... -- arguments ...
    start-stop-daemon -K|--stop options ...
    start-stop-daemon -H|--help
    start-stop-daemon -V|--version

Options (at least one of --exec|--pidfile|--user is required):
    -x|--exec <executable>       program to start/check if it is running
    -p|--pidfile <pid-file>      pid file to check
    -u|--user <username>|<uid>   stop this user's processes
    -n|--name <process-name>     start/stop processes with this name
    -s|--signal <signal>         signal to send (default 15)
    -t|--test                    test mode, don't do anything
    -o|--oknodo                  exit status 0 (not 1) if nothing done
    -q|--quiet  |  -v, --verbose

Exit status:  0 = done  1 = nothing done (=> 0 if --oknodo)  2 = trouble
```

**Pull request merged**